test/built-ins/Array/from/calling-from-valid-2.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2015 Microsoft Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+description: Calling from with a valid map function with thisArg
+es6id: 22.1.2.1
+---*/
+
+var array = [ 0, 1, -2, 4, -8, 16, -32, 64, -128, 256, -512, 1024 ];
+var arrayIndex = -1;
+var obj = {
+    mapFn (value, k) {

```

Comment:
Given that we expect implementors to be transitioning from ES5 to ES6, it's generally best to limit the number of new language features to just those that are necessary to express the behavior under test. In this case, implementations that do not support MethodDeclarations won't be able to use this test. Could you define `mapFn` using a FunctionDeclaration instead? This would also be more consistent with the related test in this patch, `calling-from-valid-1.js`